test/test_00_basic.py

```diff
@@ -34,6 +34,38 @@ def test_different_name(self):
         """, name="foo")
         assert mod.__name__ == "foo"
 
+    def test_module_typespec(self):
+        mod = self.make_module("""
+            static const HPyType_Spec Foo_spec = {
+                .name = "mytest.Foo",
+            };
+            HPyDef_TYPE(Foo_def, "Foo", Foo_spec)
```

Comment:
Clang complains that `Foo_spec` is not a compile time constant. GCC is happy with the construct. I'm not sure how to make Clang happy. I could pass the actual spec definition into the macro, but that seems likely to end up being a bit ugly.